checker/src/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java

```diff
@@ -507,8 +507,26 @@ public NullnessTypeAnnotator(InitializationAnnotatedTypeFactory<?, ?, ?, ?> atyp
 
     @Override
     public AnnotationMirror getFieldInvariantAnnotation() {
-        Elements elements = processingEnv.getElementUtils();
-        return AnnotationUtils.fromClass(elements, NonNull.class);
+        return NONNULL;
+    }
+
+    /**
+     * Returns whether or not field has the invariant annotation.
+     *
+     * <p>If the field is a type variable, this method return false if any possible instantiation of
+     * the type parameter could have the invariant annotation. In other words, is the lower
+     * bound @NonNull?
+     *
+     * @param field field that might have invariant annotation
+     * @return whether or not field has the invariant annotation
+     */
+    @Override
+    protected boolean hasInvariantAnnotation(VariableTree field) {
```

Comment:
Isn't this the implementation for all type systems? Instead of making the method abstract, how about using this implementation in the superclass? Then any future subclasses that need to change this behavior could still override the method.